frontend/src/pages/RecurringRunDetailsV2.test.tsx

```diff
@@ -0,0 +1,469 @@
+  const enableRecurringRunSpy = jest.spyOn(Apis.recurringRunServiceApi, 'enableRecurringRun');
+  const disableRecurringRunSpy = jest.spyOn(Apis.recurringRunServiceApi, 'disableRecurringRun');
+  const getExperimentSpy = jest.spyOn(Apis.experimentServiceApi, 'getExperiment');
+
+  let fullTestRecurringRun: V2beta1RecurringRun = {};
+
+  function generateProps(): PageProps {
+    const match = {
+      isExact: true,
+      params: { [RouteParams.recurringRunId]: fullTestRecurringRun.recurring_run_id },
+      path: '',
+      url: '',
+    };
+    return TestUtils.generatePageProps(
+      RecurringRunDetailsV2,
+      '' as any,
+      match,
+      historyPushSpy,
+      updateBannerSpy,
+      updateDialogSpy,
+      updateToolbarSpy,
+      updateSnackbarSpy,
+    );
+  }
+
+  beforeEach(() => {
+    fullTestRecurringRun = {
+      created_at: new Date(2018, 8, 5, 4, 3, 2),
+      description: 'test recurring run description',
+      display_name: 'test recurring run',
+      max_concurrency: '50',
+      no_catchup: true,
+      recurring_run_id: 'test-recurring-run-id',
+      runtime_config: { parameters: { param1: 'value1' } },
+      status: V2beta1RecurringRunStatus.ENABLED,
+      trigger: {
+        periodic_schedule: {
+          end_time: new Date(2018, 10, 9, 8, 7, 6),
+          interval_second: '3600',
+          start_time: new Date(2018, 9, 8, 7, 6),
+        },
+      },
+    } as V2beta1RecurringRun;
+
+    jest.clearAllMocks();
+    getRecurringRunSpy.mockImplementation(() => fullTestRecurringRun);
+    deleteRecurringRunSpy.mockImplementation();
+    enableRecurringRunSpy.mockImplementation();
+    disableRecurringRunSpy.mockImplementation();
+    getExperimentSpy.mockImplementation();
+  });
+
+  afterEach(() => {
+    if (tree) {
+      tree.unmount();
+    }
+  });
+
+  it('renders a recurring run with periodic schedule', async () => {
+    render(
+      <CommonTestWrapper>
+        <RecurringRunDetailsV2 {...generateProps()} />
+      </CommonTestWrapper>,
+    );
+
+    await waitFor(() => {
+      expect(getRecurringRunSpy).toHaveBeenCalled();
+    });
+
+    screen.getByText('Enabled');
+    screen.getByText('Yes');
+    screen.getByText('Trigger');
+    screen.getByText('Every 1 hours');
+    screen.getByText('Max. concurrent runs');
+    screen.getByText('50');
+    screen.getByText('Catchup');
+    screen.getByText('false');
+  });
+
+  it('renders a recurring run with cron schedule', async () => {
+    const cronTestRecurringRun = {
+      ...fullTestRecurringRun,
+      no_catchup: undefined, // in api requests, it's undefined when false
+      trigger: {
+        cron_schedule: {
+          cron: '* * * 0 0 !',
+          end_time: new Date(2018, 10, 9, 8, 7, 6),
+          start_time: new Date(2018, 9, 8, 7, 6),
+        },
+      },
+    };
+    getRecurringRunSpy.mockImplementation(() => cronTestRecurringRun);
+
+    render(
+      <CommonTestWrapper>
+        <RecurringRunDetailsV2 {...generateProps()} />
```

Comment:
We should use `RecurringRunDetailsRouter` instead of `RecurringRunDetailsV2` here, because RecurringRunDetailsRouter file doesn't have test coverage.